src/mon/ElectionLogic.cc

```diff
@@ -335,6 +335,10 @@ void ElectionLogic::propose_connectivity_handler(int from, epoch_t mepoch,
   ldout(cct, 10) << __func__ << " from " << from << " mepoch: "
     << mepoch << " epoch: " << epoch << dendl;
   ldout(cct, 30) << "last_election_winner: " << last_election_winner << dendl;
+  if (elector->is_stretch_marked_down_mons(from)) {
+    ldout(cct, 10) << "Ignoring proposal from marked down mon " << from << dendl;
+    return;
+  }
```

Comment:
This is supposed to be handled a few blocks below

> // drop this message; it won't change our vote so we defer to leader

Why didn't that work?

Reply:
Hi @gregsfortytwo ,
So it didn't work because there is a scenario where out-of-quorum and in-quorum mon are all both in electing state, this is due to monmap change.  Therefore, in this check:

```
if ((epoch % 2 == 0) &&
last_election_winner != elector->get_my_rank() &&
!elector->is_current_member(from))
```
We wouldn't go in because `epoch % 2 != 0` since it is in `electing` state, therefore this allows the out of quorum mon to bump the epoch of the tiebreaker mon (in quorum) since we will continue down to

``` 
if (mepoch > epoch) {
    ldout(cct, 20) << "mepoch > epoch" << dendl;
    connectivity_bump_epoch_in_election(mepoch);
```

For a full explanation with log evidence, go to `Logs analysis on a failed test` section: https://docs.google.com/document/d/1YW7RKTUlMw2nxB9vzg_7Dp32i0Tf7i1WOt6gvyZOV-g/edit

Reply:
Let's be careful with our language here: if we are electing, there is no quorum right now. :)
But that explanation makes sense.